python/llm/src/ipex_llm/transformers/models/mistral.py

```diff
@@ -1149,11 +1166,15 @@ def mistral_attention_forward_4_39_original(
         kv_seq_len += 1
 
         # update past_key_value's seem_tokens and kv caches.
-        if self.layer_idx == 0:
-            past_key_value._seen_tokens = kv_seq_len
+        # [SnapKV]
+        if use_compresskv:
+            past_key_value.update_seen_tokens(self.layer_idx, q_len)
+            kv_seq_len = past_key_value.get_seq_length()
+        else:
+            if self.layer_idx == 0:
+                past_key_value._seen_tokens = kv_seq_len
```

Comment:
Maybe we should always use `DynamicCompressCache` to make the code consistent？

Reply:
I think it's better to use another cache for example `normal cache` and implement the same interface as the compressed cache to deal with the normal case later.